core/ipc/sink/api/src/main/java/org/opennms/core/ipc/sink/api/SinkModule.java

```diff
@@ -55,15 +55,30 @@
     int getNumConsumerThreads();
 
     /**
-     * Marshals the message to a byte array.
+     * Marshals the aggregated message to a byte array.
      */
     byte[] marshal(T message);
 
     /**
-     * Unmarshals the message from a byte array.
+     * Unmarshals the aggregated message from a byte array.
      */
     T unmarshal(byte[]  message);
 
+
+    /**
+     * Marshals single message to a byte array.
+     */
+    default byte[]  marshalSingleMessage(S message) {
```

Comment:
I don't think it makes sense to provide a default impl. here - all of the implementation should define how the messages and marshaled/unmarshaled.